clang/lib/AST/ByteCode/Descriptor.h

```diff
@@ -29,6 +29,29 @@ enum PrimType : uint8_t;
 using DeclTy = llvm::PointerUnion<const Decl *, const Expr *>;
 using InitMapPtr = std::optional<std::pair<bool, std::shared_ptr<InitMap>>>;
 
+struct ArraySize {
+  unsigned Size;
```

Comment:
I find `Size` and `Capacity` to be confusing choices here. IIUC this has to do w/ the number of initializers and how many elements of the array are not initialized explicitly. These names don't really represent that at all. Maybe `NumExplicitlyInitialzed` and `ArraySize`?